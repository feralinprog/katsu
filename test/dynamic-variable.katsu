use: {
    "core.builtin.misc"
    "core.dynamic-variable"
}

let: d = (make-dynamic: "dyn")
pretty-print: d get
d set: "outer"
pretty-print: d get
d with-scope: [
    pretty-print: it
    d set: "inner"
    pretty-print: d get
]
pretty-print: d get

try: [
    d with-scope: [
        d set: "inner again"
        1 / 0
    ]
] except: {
    (Condition, \c [ print: "caught a condition" ])
}
pretty-print: d get
