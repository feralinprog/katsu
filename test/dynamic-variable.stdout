null
*string: "outer"
*string: "outer"
*string: "inner"
*string: "outer"
caught a condition
*string: "outer"
